frontend/app/routes/$lang+/_public+/apply+/$id+/communication-preference.tsx

```diff
@@ -79,49 +89,114 @@ export async function action({ request, params }: ActionFunctionArgs) {
 
   const formSchema = z
     .object({
-      preferredLanguage: z.string({ required_error: 'empty-language' }),
-      preferredMethod: z.string({ required_error: 'empty-method' }),
-      email: z.string().min(1, { message: 'empty-email' }).email({ message: 'invalid-email' }).optional(),
-      confirmEmail: z.string().min(1, { message: 'empty-confirm-email' }).optional(),
-      emailForFuture: z.preprocess((arg) => (arg === '' ? undefined : arg), z.string().email({ message: 'invalid-email' }).optional()),
-      confirmEmailForFuture: z.string().optional(),
+      preferredLanguage: z.string().min(1, { message: t('apply:communication-preference.error-message.empty-language') }),
+      preferredMethod: z.string().min(1, { message: t('apply:communication-preference.error-message.empty-method') }),
+      email: z.string().trim().optional(),
+      confirmEmail: z.string().trim().optional(),
+      emailForFuture: z.string().trim().optional(),
+      confirmEmailForFuture: z.string().trim().optional(),
     })
     .superRefine((val, ctx) => {
       if (val.preferredMethod === COMMUNICATION_METHOD_EMAIL_ID) {
+        if (isEmpty(val.email) && isEmpty(val.confirmEmail)) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.empty-confirm-email'),
+            path: ['confirmEmail'],
+          });
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.empty-email'),
+            path: ['email'],
+          });
+        }
+        if (val.email && isEmpty(val.confirmEmail)) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.empty-confirm-email'),
+            path: ['confirmEmail'],
+          });
+        }
+        if (isEmpty(val.email) && val.confirmEmail) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: t('apply:communication-preference.error-message.empty-email'),
+            path: ['email'],
+          });
+        }
```

Comment:
Maybe I'm not mentally parsing this right, but can't this be simplified to:

```
if (isEmpty(val.email)) { /* add email error */ }
if (isEmpty(val.confirmEmail)) { /* add confirm error */ }
```

I'm not seeing why all permutations of values are being checked.

Reply:
In this case, yes.  Honestly I spent a good deal of time banging my head against a wall to get this to play nice because of the optional inputs and conditional logic needed to parse.  The implementation is ugly in my mind.